runtime/parser2/type.go

```diff
@@ -734,16 +743,16 @@ func parseTypeAnnotation(p *parser) *ast.TypeAnnotation {
 
 func applyTypeNullDenotation(p *parser, token lexer.Token) ast.Type {
 	tokenType := token.Type
-	nullDenotation, ok := typeNullDenotations[tokenType]
-	if !ok {
-		panic(fmt.Errorf("unexpected token in type: %s", token.Type))
+	var nullDenotation typeNullDenotationFunc = typeNullDenotations[int(tokenType)]
```

Comment:
Use type inference, same below

```suggestion
	nullDenotation := typeNullDenotations[tokenType]
```